src/Configuration.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace JonathanTorres\Construct;
+
+use Symfony\Component\Yaml\Yaml;
+
+class Configuration
+{
+    /**
+     * Get settings derived from the configuration file.
+     *
+     * @param string                                       $configurationFile Path to the configuration file.
+     * @param string                                       $projectName       Name of the project.
+     * @param string                                       $keywords          Composer keywords.
+     * @param \JonathanTorres\Construct\Helpers\Filesystem $filesystemHelper
+     *
+     * @return \JonathanTorres\Construct\Settings
+     */
+    public static function getSettings($configurationFile, $projectName, $keywords, $filesystemHelper)
+    {
+        if (!$filesystemHelper->isFile($configurationFile)) {
+            $exceptionMessage = "Configuration file '$configurationFile' is not existent.";
+            throw new \RuntimeException($exceptionMessage);
+        }
+
+        if (!$filesystemHelper->isReadable($configurationFile)) {
+            $exceptionMessage = "Configuration file '$configurationFile' is not readable.";
+            throw new \RuntimeException($exceptionMessage);
+        }
+
+        $configuration = Yaml::parse($filesystemHelper->get($configurationFile));
+
+        if (isset($configuration['construct-with'])) {
+            $configuration['construct-with'] = array_flip($configuration['construct-with']);
+        }
+
+        return new Settings(
+            $projectName,
+            isset($configuration['test-framework']) ? $configuration['test-framework'] : (new Defaults())->testingFrameworks[0],
```

Comment:
rather than creating 2 `Defaults` objects, could you instantiate one and use it here?